timely/src/logging.rs

```diff
@@ -46,13 +46,26 @@ impl<T, E, P> Drop for BatchLogger<T, E, P> where P: EventPusher<Duration, (Dura
     }
 }
 
+#[derive(Serialize, Deserialize, Abomonation, Debug, Clone, Hash, Eq, PartialEq, Ord, PartialOrd)]
+/// The creation of a `Subgraph`.
+pub struct SubgraphEvent {
+    /// Worker-unique identifier for the operator.
+    pub id: usize,
+    /// Sequence of nested scope identifiers indicating the path from the root to this instance.
+    pub addr: Vec<usize>,
+    /// Type of the subscope Timestamp.
+    pub timestamp_type: String,
+}
+
 #[derive(Serialize, Deserialize, Abomonation, Debug, Clone, Hash, Eq, PartialEq, Ord, PartialOrd)]
 /// The creation of an `Operate` implementor.
 pub struct OperatesEvent {
     /// Worker-unique identifier for the operator.
     pub id: usize,
     /// Sequence of nested scope identifiers indicating the path from the root to this instance.
     pub addr: Vec<usize>,
+    /// Internal summary for every combination of input and output port.
+    pub internal_summaries: Vec<Vec<String>>,
```

Comment:
Can you explain what these are used for? Would it be equally beneficial to have the report from the tracker about its input-to-output summaries? That would leave this event stable and consolidate the timestamp/summary related events to the reachability tracker.